pkg/service/aws/oidc.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/pkg/errors"
+	"gitlab.com/netbook-devs/spawner-service/pkg/config"
+	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
+)
+
+//arn:aws:iam::965734315247:oidc-provider
+//
+// oidc endpoint
+const trustPolicTemplate = `{
+			"Effect": "Allow",
+			"Principal": {
+				"Federated": "%s"
+			},
+			"Action": "sts:AssumeRoleWithWebIdentity",
+			"Condition": {
+				"StringEquals": {
+					"%s:sub": "system:serviceaccount:oidc:awskube2iam",
```

Comment:
@FiReBeNdEr07 can you confirm the service account namespace and name?
@NishanthSpShetty can we pick the namespace `oidc` and service account name `awskube2iam` from constants?

Reply:
updated.